src/libs/actions/connections/index.ts

```diff
@@ -0,0 +1,113 @@
+import Onyx from 'react-native-onyx';
+import type {OnyxUpdate} from 'react-native-onyx';
+import type {ValueOf} from 'type-fest';
+import * as API from '@libs/API';
+import {WRITE_COMMANDS} from '@libs/API/types';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {PolicyConnectionName} from '@src/types/onyx/Policy';
+
+function removePolicyConnection(policyID: string, connectionName: PolicyConnectionName) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    [connectionName]: null,
+                },
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.POLICY_CONNECTION_SYNC_PROGRESS}${policyID}`,
+            value: null,
+        },
+    ];
+    const failureData: OnyxUpdate[] = [
+        // {
+        //     onyxMethod: Onyx.METHOD.MERGE,
+        //     key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+        //     value: {
+        //         errorFields: {
+        //             avatar: ErrorUtils.getMicroSecondOnyxError('avatarWithImagePicker.deleteWorkspaceError'),
+        //         },
+        //     },
+        // },
+    ];
+    API.write(WRITE_COMMANDS.REMOVE_POLICY_CONNECTION, {policyID, connectionName}, {optimisticData, failureData});
+}
+
+function updatePolicyConnectionConfig(policyID: string, settingName: ValueOf<typeof CONST.QUICK_BOOKS_IMPORTS>, settingValue: ValueOf<typeof CONST.INTEGRATION_ENTITY_MAP_TYPES>) {
+    const parameters = {policyID, connectionName: CONST.QUICK_BOOKS_ONLINE, settingName, settingValue, idempotencyKey: settingName};
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    quickbooksOnline: {
+                        config: {
+                            [settingName]: settingValue,
+                            pendingFields: {
+                                [settingName]: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                            },
+                            errorFields: {
+                                [settingName]: null,
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    quickbooksOnline: {
+                        config: {
+                            [settingName]: settingValue,
```

Comment:
True, this is wrong. I'm now regretting moving this function from [policy.ts](https://github.com/Expensify/App/pull/39219/files#diff-cd0dfd9f8af5bc7cd33ff4f0337ef70574d42424ceecd7692b193348d3d8ecd4L3724-L3794) in the context of this PR.

I would like to fix `updatePolicyConnectionConfig` in another PR since it is out of the scope for the Authentication and these bugs were already there. Sorry about moving it here, it made the PR messier.

Reply:
that was my fault that I made mistake with the function in my commit

anyway, I agree. let's fix it in a follow up PR.